### PR TITLE
Det and seg visualization & bug fixes

### PR DESCRIPTION
**Pull request by will-jl944**
1. visualization for seg and det
2. consider the special case of detection evaluation in get_single_card_bn (only single card with batch size 1 is supported)
3. det evaluation supports either voc metric or coco metric, regardless the dataset format
4. CocoDetection format also supports object detection task
5. remove unique_name_guard in segmenter
6. add draw_pr_curve

### CODE REVIEW

File: paddlex/cv/models/detector.py [hunk 1]
```diff
@@ -237,28 +267,38 @@ def evaluate(self, eval_dataset, batch_size, return_details=False):
...
                         labels=eval_dataset.labels,
                         is_bbox_normalized=is_bbox_normalized,
                         classwise=False)
-                ]
-            elif eval_dataset.__class__.__name__ == 'CocoDetection':
-                eval_metrics = [
-                    COCOMetric(
+                elif eval_dataset.__class__.__name__ == 'CocoDetection':
+                    eval_metric = COCOMetric(
                         coco_gt=eval_dataset.coco_gt, classwise=False)
```
Comment (FlyingQianMM) [inline paddlex/cv/models/detector.py]: check下eval_dataset.coco_gt是否需要做deepcopy

File: paddlex/cv/models/utils/det_metrics/metrics.py [hunk 2]
```diff
@@ -195,10 +209,22 @@ def accumulate(self):
...
             sys.stdout.flush()
 
+        output_path = 'eval_result.json'
+        if self.output_eval:
+            output_path = os.path.join(self.output_eval, output_path)
+        with open(output_path, 'w') as f:
+            json.dump(self.results, f)
+            logging.info('The evaluation result is saved at {}.'.format(
+                output_path))
+
```
Comment (FlyingQianMM) [inline paddlex/cv/models/utils/det_metrics/metrics.py]: 这个不用保存在output_eval里面，直接保存在每个epoch的save_dir里面，像静态图这样 https://github.com/PaddlePaddle/PaddleX/blob/ad667c23694b48b3e83d9b62068dcf3bdbdc2d43/paddlex/cv/models/base.py#L359 。eval_result.json实际是以coco格式保存的json文件，还要保存真值: https://github.com/PaddlePaddle/PaddleX/blob/ad667c23694b48b3e83d9b62068dcf3bdbdc2d43/paddlex/cv/models/utils/detection_eval.py#L51 这样通过eval_result.json可以重新得到评估精度。

File: paddlex/cv/nets/paddleseg/models/layers/activation.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ def __init__(self, act=None):
         super(Activation, self).__init__()
 
         self._act = act
-        upper_act_names = nn.layer.activation.__all__
+        upper_act_names = nn.layer.activation.__dict__.keys()
         lower_act_names = [act.lower() for act in upper_act_names]
```
Comment (FlyingQianMM) [inline paddlex/cv/nets/paddleseg/models/layers/activation.py:57]: 不要直接修改paddleseg的代码

File: paddlex/cv/models/utils/det_metrics/metrics.py [hunk 4]
```diff
@@ -77,6 +77,7 @@ def __init__(self,
         self.reset()
 
     def reset(self):
+        self.results = {'bbox': []}
```
Comment (FlyingQianMM) [inline paddlex/cv/models/utils/det_metrics/metrics.py]: self.results和下面的self.get_results()其中一个改下名字，跟COCOMetric统一

File: paddlex/cv/models/detector.py (hunk 1 above)
Comment (will-jl944) [reply paddlex/cv/models/detector.py]: Done

File: paddlex/cv/models/utils/det_metrics/metrics.py (hunk 2 above)
Comment (will-jl944) [reply paddlex/cv/models/utils/det_metrics/metrics.py]: Doing

File: paddlex/cv/models/utils/det_metrics/metrics.py (hunk 4 above)
Comment (will-jl944) [reply paddlex/cv/models/utils/det_metrics/metrics.py]: Done

File: paddlex/cv/models/utils/det_metrics/metrics.py (hunk 2 above)
Comment (will-jl944) [reply paddlex/cv/models/utils/det_metrics/metrics.py]: Done

**Pull request merged**